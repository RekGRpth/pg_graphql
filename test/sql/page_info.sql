begin;
    create table account(
        id int primary key,
        is_verified bool
    );


    -- hasNextPage and hasPreviousPage should be non-null on empty collection
    -- startCursor and endCursor may be null
    select jsonb_pretty(
        graphql.resolve($$
        {
          accountCollection {
            pageInfo {
              hasNextPage
              hasPreviousPage
              startCursor
              endCursor
            }
          }
        }
        $$)
    );
