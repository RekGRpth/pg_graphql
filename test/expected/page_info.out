begin;
    create table account(
        id int primary key,
        is_verified bool
    );
    -- hasNextPage and hasPreviousPage should be non-null on empty collection
    -- startCursor and endCursor may be null
    select jsonb_pretty(
        graphql.resolve($$
        {
          accountCollection {
            pageInfo {
              hasNextPage
              hasPreviousPage
              startCursor
              endCursor
            }
          }
        }
        $$)
    );
               jsonb_pretty               
------------------------------------------
 {                                       +
     "data": {                           +
         "accountCollection": {          +
             "pageInfo": {               +
                 "endCursor": null,      +
                 "hasNextPage": false,   +
                 "startCursor": null,    +
                 "hasPreviousPage": false+
             }                           +
         }                               +
     },                                  +
     "errors": [                         +
     ]                                   +
 }
(1 row)

